components/automate-chef-io/README.md

```diff
@@ -81,11 +88,314 @@ This will create the file and fill in the necessary frontmatter automatically.
+  This is the description.
+  The description can span multiple lines and can contain `Github Flavored Markdown`
+  */
+  rpc ListReports(Query) returns (Reports) {
+    option (google.api.http) = {
+      post: "/compliance/reporting/reports"
+      body: "*"
+    };
+    option (grpc.gateway.protoc_gen_swagger.options.openapiv2_operation) = {
+      tags: "Reporting"
+    };
+  };
+```
+
+### Attribute Style Descriptions
+
+```go
+rpc ListReports(Query) returns (Reports) {
+    option (google.api.http) = {
+      post: "/compliance/reporting/reports"
+      body: "*"
+    };
+    option (grpc.gateway.protoc_gen_swagger.options.openapiv2_operation) = {
+      tags: "Reporting"
+      summary: "List reports"
+      description: "This is the description.\nThe description can span multiple lines and can contain Github Flavored Markdown."
+    };
+  };
+```
+
+Resource documentation provides RPC-level description. It is sometimes also called "method description" because it documents the what you can do with a specific API HTTP Method.
+
+Use block/leading comments for RPC-level/method documentation.
+
+Compare the leading comments to the content of the API documentation for the `ReadNode(ID)` resource.
+
+```go
+	/*
+	Fetch a node #(summary)
+
+	Fetch a specific node by id. #(description)
+	Does not support filtering, pagination or sorting.
+	*/
+	rpc ReadNode(Id) returns (Node) {
+```
+
+The rendered documentation:
+
+![](./static/images/docs/api-resource-definition.png)
+
+### Message Level Descriptions
+
+Document fields and parameters directly above the relevant code line, using `//` inline comments
+
+## Message Documentation
+
+Given messages defined like this in the proto files:
+
+```go
+message Dog {
+  // The dog's name.
+  string name = 1;
+  // Intentionally blank.
+  Food favorite_food = 2;
+  // Types of food the dog is normally fed.
+  repeated Food typical_foods = 3;
+}
+// Stats on the statuses of nodes matching the filters.
+message ProfileCounts {
+  // The food's name.
+  string name = 1;
+  // Serving size in ounces.
+  int32 serving_size =3;
+}
+```
+
+* Use a period (`.`) at the end of the comment controls whether the field will get to set the `description` for message fields.
+* Swagger relies on inheritance to document nested references. A parent message referenced in a child message inherits its description from the parent, not where it is referenced by the child(see the example of `favorite_food` above). The Chef API style uses `// Intentionally blank.` in the child message.
+This convention makes the purpose of the comment clear, makes undocumented message code visible, and makes bad blank message comment lines stand out in the UI.
+
+## Non-Proto File Sources for API Documentation
+
+We use non-proto file sources to document API content in cases where proto file conversion to Swagger our the front-end ReDoc tools have pronounced limitations, as well as in cases where expressing the content in proto files is cumbersome.
```

Comment:
Clarification is needed for this sentence. It's possible this should be two sentences, but I am not sure.